features/netsocket/TCPSocket.cpp

```diff
@@ -22,9 +22,19 @@ TCPSocket::TCPSocket()
 {
 }
 
+TCPSocket::TCPSocket(TCPSocket* parent, nsapi_socket_t socket, SocketAddress address)
+{
+    _socket = socket,
+    _stack = parent->_stack;
+    _factory_allocated = true;
+    _remote_peer = address;
```

Comment:
Would it actually compile in less code, if we would use these values in initializer?

```
TCPSocket::TCPSocket(TCPSocket* parent, nsapi_socket_t socket, SocketAddress address) :
    _socket(socket),
    _stack(parent->_stack),
    _factory_allocated(true),
    _remote_peer(address)
{
    ...
```

At least it would use copy constructor for peer address.

Reply:
We cannot initialise members of the base class in the constructor initialisation list of the derived class. 
We would have to add a dedicated constructor for InternetSocket and then call:
TCPSocket::TCPSocket (...) : InternetSocket(socket, parent->_stack, ...) {}
I though this was to much of interference with the API, but if you think this is acceptable I can add the constructor.